src/pages/bus_checkin_roster/[route].jsx

```diff
@@ -338,50 +346,57 @@ const Roster = () => {
   }
 
   return (
-    <div className={classes.container}>
-      <div className={classes.header}>
-        <Link href="/route_selection">
-          <button className={classes.backbtn}>
-            <ArrowBackIosIcon />
-            <h1 className={classes.text}>Back </h1>
-          </button>
-        </Link>
-        <h1>{route}</h1>
-      </div>
-      <table style={{ width: "100%" }}>
-        <thead>
-          <tr className={classes.tr}>
-            <th className={classes.th} style={{ width: "25%" }}>
-              Name
-            </th>
-            <th className={classes.th}>Status</th>
-          </tr>
-        </thead>
-        <tbody className={classes.tbody}>
-          {students.map((student, index) => (
-            <tr className={classes.tr} key={index}>
-              <td className={classes.td} style={{ width: "25%" }}>
-                {student.name}
-              </td>
-              {student.checkedIn ? (
-                <StudentCheckedIn
-                  justCheckedIn={student.note == ""}
-                  index={index}
-                />
-              ) : (
-                <StudentNotCheckedIn index={index} />
-              )}
-            </tr>
-          ))}
-        </tbody>
-      </table>
-      <ModalComponent
-        button={<h1 style={{ margin: "0px" }}>Submit</h1>}
-        buttonStyle={classes.mainSubmitBtn}
-      >
-        <SubmitModalContent />
-      </ModalComponent>
-    </div>
+    <>
+      {session && userType === "BusDriver" ? (
```

Comment:
Instead of putting this logic into the returned component, pull the conditional into an if that returns nothing.